src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -283,6 +306,128 @@ public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entity
+        /// <param name="restPathsForEntities">Set of unique rest paths configured for the entities in the config.</param>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private static void ValidateRestPathForEntity(string entityName, JsonElement restPathElement, HashSet<string> restPathsForEntities)
+        {
+            if (restPathElement.ValueKind is JsonValueKind.Null)
+            {
+                // The rest path can't be null.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has a null rest {RestEntitySettings.PROPERTY_PATH}. Accepted value types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is not JsonValueKind.True && restPathElement.ValueKind is not JsonValueKind.False
+                && restPathElement.ValueKind is not JsonValueKind.String)
+            {
+                // The rest path can only be a string or a boolean value.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has rest {RestEntitySettings.PROPERTY_PATH} specified with incorrect data type. " +
+                    $"Accepted data types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is JsonValueKind.String)
+            {
+                string path = restPathElement.ToString();
+                if (string.IsNullOrEmpty(path))
+                {
+                    // The rest 'path' cannot be empty.
+                    throw new DataApiBuilderException(
+                        message: $"Entity: {entityName} has an empty rest {RestEntitySettings.PROPERTY_PATH}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                if (restPathsForEntities.Contains(path))
+                {
+                    // Presence of multiple entities having the same rest path configured causes conflict.
+                    throw new DataApiBuilderException(
+                        message: $"Multiple entities found with same rest {RestEntitySettings.PROPERTY_PATH}: {path}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                restPathsForEntities.Add(path);
```

Comment:
```suggestion
                if (!restPathsForEntities.Add(path))
                {
                    // Presence of multiple entities having the same rest path configured causes conflict.
                    throw new DataApiBuilderException(
                        message: $"Multiple entities found with same rest {RestEntitySettings.PROPERTY_PATH}: {path}.",
                        statusCode: HttpStatusCode.ServiceUnavailable,
                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
                        );
                }
```

possible simplification

Reply:
Wow, I was looking for something like `TryAdd`. Wasn't aware that `Add` does it for us here. Thanks!